pkg/asset/installconfig/kubevirt/validation.go

```diff
@@ -0,0 +1,120 @@
+package kubevirt
+
+import (
+	"errors"
+	"fmt"
+	"net"
+
+	"github.com/openshift/installer/pkg/types"
+	"github.com/openshift/installer/pkg/types/kubevirt"
+	"github.com/openshift/installer/pkg/types/kubevirt/validation"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// Validate executes kubevirt specific validation
+func Validate(ic *types.InstallConfig, clientBuilder ClientBuilderFuncType) error {
+	kubevirtPlatformPath := field.NewPath("platform", "kubevirt")
+
+	if ic.Platform.Kubevirt == nil {
+		return errors.New(field.Required(
+			kubevirtPlatformPath,
+			"validation requires a KubeVirt platform configuration").Error())
+	}
+
+	return validatePlatform(ic.Platform.Kubevirt, ic.MachineNetwork, kubevirtPlatformPath, clientBuilder).ToAggregate()
+}
+
+func validatePlatform(kubevirtPlatform *kubevirt.Platform, machineNetworkEntryList []types.MachineNetworkEntry, fldPath *field.Path, clientBuilder ClientBuilderFuncType) field.ErrorList {
+	allErrs := validation.ValidatePlatform(kubevirtPlatform, fldPath)
+
+	client, resultErrs := validateInfraClusterReachable(kubevirtPlatform.Namespace, fldPath, clientBuilder)
+	allErrs = append(allErrs, resultErrs...)
+	if client != nil {
+		allErrs = append(allErrs, validateStorageClassExistsInInfraCluster(kubevirtPlatform.StorageClass, client, fldPath)...)
+		allErrs = append(allErrs, validateNetworkAttachmentDefinitionExistsInNamespace(kubevirtPlatform.NetworkName, kubevirtPlatform.Namespace, client, fldPath)...)
+	}
+	allErrs = append(allErrs, validateIPsInMachineNetworkEntryList(machineNetworkEntryList, kubevirtPlatform.APIVIP, kubevirtPlatform.IngressVIP, fldPath)...)
+
+	return allErrs
+}
+
+// validateInfraClusterReachable validates the following:
+// 1. Client can be created -  login to cluster is done or KUBECONFIG environment variable was exported
+// 2. Cluster is reachable - valid user
+// 3. Kubevirt installed
+// 4. User can list VMs in namespace
+func validateInfraClusterReachable(namespace string, fieldPath *field.Path, clientBuilder ClientBuilderFuncType) (Client, field.ErrorList) {
```

Comment:
The install config has nothing to do with cluster reachability. As such, the reachability check should not be included in validation. Moreover, the validation code should be passed in a client rather than having to create one.

Reply:
Fixed: Removed this validation and client is passed from caller